proof_of_stake/src/epoched.rs

```diff
@@ -269,8 +265,13 @@ where
         LazyMap::open(key)
     }
 
-    fn sub_past_epochs(epoch: Epoch) -> Epoch {
-        Epoch(epoch.0.checked_sub(NUM_PAST_EPOCHS).unwrap_or_default())
+    fn sub_past_epochs(params: &PosParams, epoch: Epoch) -> Epoch {
+        Epoch(
+            epoch
+                .0
+                .checked_sub(PastEpochs::value(params))
+                .unwrap_or_default(),
+        )
```

Comment:
nit: not really in the scope of this pr I guess, but `Epoch` is a good candidate for a `checked_sub` method, so you don't have to do these destruct-construct gymnastics to subtract a value safely

Reply:
Yeah, that's shout! We do have `checked_sub` but it takes `Epoch` for right-hand side. We should change it to take `impl Into<Epoch>` to be able to use it with e.g. `u64`

Reply:
I've done this in [2e15d51](https://github.com/anoma/namada/pull/1944/commits/2e15d519604952e54eb365c0e3425df5bbf1f526) and [d3048b3](https://github.com/anoma/namada/pull/1944/commits/d3048b32114920c42a099e1009fb34ed7c6cd19a) - it's a non-breaking change so it's fine to add here imho